// xgplayer-service-miniprogram/webpack.config.js
const polyfill = [];
var webpack = require('webpack');

const umd = {
    target: 'node',
    entry: polyfill.concat(['./src/index.js']),
    output: {
        path: __dirname + '/dist',
        filename: 'index.js',
        library: 'xgplayer-service',
        libraryTarget: 'commonjs2',
    },
    mode: 'production',
    node: {
        process: false,
    },
    module: {
        rules: [{
            test: /\.js$/,
            loader: 'babel-loader',
        }],
    },
    optimization: {
        minimize: true,
    },
    plugins: [
        new webpack.DefinePlugin({
            'process.env.NODE_ENV': JSON.stringify(process.env.NODE_ENV),
        }),
    ],
};

const client = {
    entry: polyfill.concat(['./src/index.js']),
    output: {
        path: __dirname + '/browser',
        filename: 'index.js',
        library: 'XgplayerService',
        libraryTarget: 'window',
    },
    module: {
        rules: [{
            test: /\.js$/,
            loader: 'babel-loader',
        }],
    },
    mode: 'production',
    optimization: {
        minimize: true,
    },
};

module.exports = [umd, client];
